### DISCUSSION TITLE
How to specify the sort method for a column in a sortable table

### DISCUSSION THREAD

**Discussion by Stephen-Gates** [squidfunk/mkdocs-material#3577] [Q&A] [answered]
I've implemented table sorting as described in the [documentation](https://squidfunk.github.io/mkdocs-material/reference/data-tables/#sortable-tables). The [sort type](https://tristen.ca/tablesort/demo/#sort-types) defaults to sort by string. [Other sort types](https://github.com/tristen/tablesort/tree/gh-pages/src/sorts/) are available.

I'm seeking help on how to [specify the sort method for a column](https://tristen.ca/tablesort/demo/#specify-the-sort-method-for-a-column). E.g. How do I add the `data-sort-method` parameter below, to the Markdown table header `| Version |` to achieve:

`<th data-sort-method='dotsep'>Version</th>`

**Comment by squidfunk** [comment, marked as answer]
Unfortunately, it's not possible to add attributes to tables, which is a limitation of the [Attribute Lists extension](https://python-markdown.github.io/extensions/attr_list/#limitations):

> There are various HTML elements which are not represented in Markdown text, but only implied. For example, the ul and ol elements do not exist in Markdown. They are only implied by the presence of list items (li). **There is no way to use an attribute list to define attributes on implied elements, including but not limited to the following: ul, ol, dl, table, thead, tbody, and tr**.

You would need to provide your table as raw HTML in order to specify attributes. Other than that, you could also use JavaScript to add the class to the corresponding column before the table sort code is executed since table sort needs JavaScript anyway.

**Comment by elliot-100** [comment]
[Deleted as it did not add any value - issue came down to (probably buggy) tablesort behaviour]
